### PR TITLE
Add nightly for each repository release stream

### PR DESCRIPTION
**Pull request by ehelms**
This change will allow controlling for what OSes a given release from our release jobs, for nightly, and align it with a numerical release. Right now, we store in Jenkins the OSes nightly supports and then we iterate and call the `build_stage_repository` script [1]. By having the nightly releases captured here, our CI can instead call `VERSION='nightly' PROJECT=<project> ./generate_stage_repositories` and allow configuration inside this repository to drive what is generated.

In a similar fashion, we can make use of a single rsync script to stagingyum to handle uploads for nightly and releases in the same way without having to keep CI and this repository in sync. See https://github.com/theforeman/jenkins-jobs/pull/369.

[1] https://github.com/theforeman/jenkins-jobs/blob/master/theforeman.org/pipelines/release/pipelines/foreman-rpm.groovy#L69-L73

### CODE REVIEW

**Review by ekohl — CHANGES_REQUESTED** [review]
We don't have rc1 in nightly. I'm not sure what a valid `FULLVERSION` would be. For example, in our release pipeline we pass `FULLVERSION` as the expected version and that would mean it expects `nightly.0-rc1` in `/usr/share/foreman*/VERSION`. Perhaps it should just not be set at all?

**Comment by ehelms** [comment]
> We don't have rc1 in nightly. I'm not sure what a valid `FULLVERSION` would be. For example, in our release pipeline we pass `FULLVERSION` as the expected version and that would mean it expects `nightly.0-rc1` in `/usr/share/foreman*/VERSION`. Perhaps it should just not be set at all?

For now I shortened it to nightly. Right now this appears to be forced by https://github.com/theforeman/theforeman-rel-eng/blob/master/settings#L25-L29 and I wouldn't want to relax this for releases. Whereas having this ability to use scripts in all versions including nightly is strong for centralizing our scripts. Perhaps we take this version as is and keep iterating if FULLVERSIOn for nightly becomes tricky with other workflows?

**Comment by ekohl** [comment]
In https://github.com/theforeman/theforeman-rel-eng/pull/330 I made an effort to make some variables optional, which makes the files more logical.

**Review by ekohl — COMMENTED** [review]
I think we can now drop `FULLVERSION` in all of these files. The flows that need it shouldn't run and otherwise properly error out.

**Comment by ehelms** [comment]
> I think we can now drop `FULLVERSION` in all of these files. The flows that need it shouldn't run and otherwise properly error out.

Cleaned up, and seems to be working from my test scenarios on scripts.

File: releases/client/nightly/settings [hunk 1]
```diff
@@ -0,0 +1 @@
+FOREMAN_VERSION='nightly'
```
Comment (ekohl) [inline releases/client/nightly/settings:1]: We set this by default:
https://github.com/theforeman/theforeman-rel-eng/blob/6a9a8a39e3b0a4889102c7add4132a624a1ef5cb/releases/client/settings#L3
```suggestion
```

File: releases/candlepin/nightly/settings [hunk 2]
```diff
@@ -0,0 +1 @@
+FOREMAN_VERSION='nightly'
```
Comment (ekohl) [inline releases/candlepin/nightly/settings:1]: Candlepin doesn't have a Foreman version:
https://github.com/theforeman/theforeman-rel-eng/blob/6a9a8a39e3b0a4889102c7add4132a624a1ef5cb/releases/candlepin/settings#L4

```suggestion
```

File: releases/plugins/nightly/settings [hunk 3]
```diff
@@ -0,0 +1 @@
+FOREMAN_VERSION='nightly'
```
Comment (ekohl) [inline releases/plugins/nightly/settings:1]: I'd prefer a solution similar to `client` where in `releases/plugins/settings` we set `FOREMAN_VERSION=$VERSION` and be done with it.
```suggestion
```

File: releases/plugins/settings [hunk 4]
```diff
@@ -0,0 +1,2 @@
+PACKAGING_SUBDIR="packages/plugins"
+OSES="el8"
```
Comment (ekohl) [inline releases/plugins/settings:2]: As said above: we can set it by default (and sort the file alphabetically):
```suggestion
FOREMAN_VERSION=$VERSION
OSES="el8"
PACKAGING_SUBDIR="packages/plugins"
```

**Review by ekohl — CHANGES_REQUESTED** [review]
I should have caught this in the previous review, but more simplifications are possible.

**Review by ekohl — CHANGES_REQUESTED** [review]
Wouldn't you still need a `release/plugins/nightly/settings` file, since it will error out if it doesn't exist? By now that needs `OSES="el8 el9"`.

File: releases/pulpcore/nightly/settings [hunk 5]
```diff
@@ -0,0 +1 @@
+FULLVERSION='nightly.0-rc1'
```
Comment (ekohl) [inline releases/pulpcore/nightly/settings:1]: Pulpcore doesn't have a full version either:
```suggestion
```

**Review by ekohl — CHANGES_REQUESTED** [review]
Last one, I promise ;)

File: releases/client/settings [hunk 6]
```diff
@@ -1,3 +1,4 @@
+FOREMAN_VERSION=$VERSION
```
Comment (evgeni) [inline releases/client/settings:1]: This line now exists twice in this file

**Review by evgeni — CHANGES_REQUESTED** [review]
Glad *I* didn't promise anything

File: releases/client/settings (hunk 6 above)
Comment (ehelms) [reply releases/client/settings:1]: Snuck right in there! Also reading comments out of order is both confusing and fun.

**Review by ekohl — APPROVED** [review]

**Pull request merged**